### PR TITLE
feat: Functionality added to install chart and upgrade chart with provided chart repo

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
# Description
Two APIs : 
1) Install chart 
2) Upgrade chart with provided chart repository

Fixes # https://github.com/devtron-labs/devtron/issues/1151

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
New chart install and chart upgrade. (stateless check)

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: pkg/builder/AppDetailBuilder.go [hunk 1]
```diff
@@ -384,6 +384,130 @@ func GetDeploymentDetail(request *client.DeploymentDetailRequest) (*client.Deplo
 	return resp, nil
 }
```
Comment (nishant-d) [inline pkg/builder/AppDetailBuilder.go:385]: change file name to represent current functionality

File: pkg/builder/AppDetailBuilder.go [hunk 2]
```diff
@@ -384,6 +384,130 @@ func GetDeploymentDetail(request *client.DeploymentDetailRequest) (*client.Deplo
...
+		return nil, err
+	}
+	// Add or update chart repo ends
+
+	// Install release starts
+	chartSpec := &helmClient.ChartSpec{
+		ReleaseName:      releaseIdentifier.ReleaseName,
+		Namespace:        releaseIdentifier.ReleaseNamespace,
+		ValuesYaml:       request.ValuesYaml,
+		ChartName:        chartRepoName + "/" + request.ChartName,
```
Comment (nishant-d) [inline pkg/builder/AppDetailBuilder.go:429]: use fmtf instead of string concatenation

File: pkg/builder/helmAppService.go [hunk 3]
```diff
@@ -101,6 +106,20 @@ func (impl *ApplicationServiceServerImpl) GetDeploymentDetail(ctx context.Contex
 	return GetDeploymentDetail(in)
 }
 
+func (impl *ApplicationServiceServerImpl) InstallRelease(ctx context.Context, in *client.InstallReleaseRequest) (*client.InstallReleaseResponse, error) {
+	impl.logger.Infow("install release request")
```
Comment (nishant-d) [inline pkg/builder/helmAppService.go:110]: add some contextual info in log

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: pkg/builder/AppDetailBuilder.go (hunk 1 above)
Comment (manish-agrawal-ai) [reply pkg/builder/AppDetailBuilder.go:385]: Will do this in my next item (Refactoring #1152)

File: pkg/builder/AppDetailBuilder.go [hunk 4]
```diff
@@ -384,6 +384,130 @@ func GetDeploymentDetail(request *client.DeploymentDetailRequest) (*client.Deplo
...
+		Version:          request.ChartVersion,
+		DependencyUpdate: true,
+		UpgradeCRDs:      true,
+	}
+	_, err = helmClientObj.InstallChart(context.Background(), chartSpec)
```
[the remark this answers is not in the document]
Comment (manish-agrawal-ai) [reply pkg/builder/AppDetailBuilder.go:435]: Will do this in my next item (Refactoring #1152) as this is applicable for all

File: pkg/builder/helmAppService.go (hunk 3 above)
Comment (manish-agrawal-ai) [reply pkg/builder/helmAppService.go:110]: Will do this in my next item (Refactoring #1152)

File: pkg/helmClient/client.go [hunk 5]
```diff
@@ -144,6 +146,54 @@ func (c *HelmClient) UpgradeRelease(ctx context.Context, chart *chart.Chart, upd
...
+// InstallChart installs the provided chart and returns the corresponding release.
+// Namespace and other context is provided via the helmclient.Options struct when instantiating a client.
+func (c *HelmClient) InstallChart(ctx context.Context, spec *ChartSpec) (*release.Release, error) {
+	installed, err := c.chartIsInstalled(spec)
+	if err != nil {
+		return nil, err
+	}
+
+	if installed {
+		return nil, errors.New("release already exists with this releaseName and namespace")
```
[the remark this answers is not in the document]
Comment (manish-agrawal-ai) [reply pkg/helmClient/client.go:186]: done

File: pkg/builder/AppDetailBuilder.go (hunk 2 above)
Comment (manish-agrawal-ai) [reply pkg/builder/AppDetailBuilder.go:429]: done

File: pkg/builder/helmAppService.go [hunk 6]
```diff
@@ -101,6 +106,20 @@ func (impl *ApplicationServiceServerImpl) GetDeploymentDetail(ctx context.Contex
...
+	impl.chartRepositoryLocker.Lock(in.ChartRepository.Name)
+	defer impl.chartRepositoryLocker.Unlock(in.ChartRepository.Name)
+	return InstallRelease(in)
+}
+
+func (impl *ApplicationServiceServerImpl) UpgradeReleaseWithChartInfo(ctx context.Context, in *client.InstallReleaseRequest) (*client.UpgradeReleaseResponse, error) {
+	impl.logger.Infow("upgrade release with chart info request")
+	impl.chartRepositoryLocker.Lock(in.ChartRepository.Name)
+	defer impl.chartRepositoryLocker.Unlock(in.ChartRepository.Name)
```
[the remark this answers is not in the document]
Comment (manish-agrawal-ai) [reply pkg/builder/helmAppService.go:119]: invalid

File: pkg/builder/AppDetailBuilder.go (hunk 1 above)
Comment (manish-agrawal-ai) [reply pkg/builder/AppDetailBuilder.go:385]: done

File: pkg/builder/AppDetailBuilder.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (manish-agrawal-ai) [reply pkg/builder/AppDetailBuilder.go:435]: done

**Review by nishant-d — APPROVED** [review]

**Pull request merged**